.github/workflows/create-source-binary.yml

```diff
@@ -30,41 +24,41 @@ jobs:
   ##  - workflow is building default branch (master)
   artifact:
     if: |
-      inputs.tag != '' && 
+      inputs.tag != '' &&
       github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
     name: Build Binaries
     runs-on: ubuntu-latest
     strategy:
       ## Run a maximum of 10 builds concurrently, using the matrix defined in inputs.arch
       max-parallel: 10
       matrix:
-        platform: ${{ fromJson(inputs.arch) }}
+        arch:
+          - linux-musl
+          - linux-glibc
+          - macos
+          - windows
+        cpu:
+          - arm64
+          - armv7
+          - x86-64 # defaults to x86-64-v3 variant - intel haswell (2013) and newer
+          # - x86-64-v2 ## intel nehalem (2008) and newer
+          # - x86-64-v3 ## intel haswell (2013) and newer
+          # - x86-64-v4 ## intel skylake (2017) and newer
+        exclude:
+          - arch: windows # excludes windows-arm64
+            cpu: arm64
+          - arch: windows # excludes windows-armv7
+            cpu: armv7
+          - arch: macos # excludes macos-armv7
+            cpu: armv7
+    env:
+      TEST_TIMEOUT: 30
     steps:
-      ## Setup Docker for the builds
-      - name: Docker setup
-        uses: stacks-network/actions/docker@main
-
-      ## Build the binaries using defined dockerfiles
-      - name: Build Binary (${{ matrix.platform }})
-        id: build_binaries
-        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 # 5.0.0
-        with:
-          file: build-scripts/Dockerfile.${{ matrix.platform }}
-          outputs: type=local,dest=./release/${{ matrix.platform }}
-          build-args: |
-            STACKS_NODE_VERSION=${{ inputs.tag || env.GITHUB_SHA_SHORT }}
-            OS_ARCH=${{ matrix.platform }}
-            GIT_BRANCH=${{ env.GITHUB_REF_SHORT }}
-            GIT_COMMIT=${{ env.GITHUB_SHA_SHORT }}
-
-      ## Compress the binary artifact
-      - name: Compress artifact
-        id: compress_artifact
-        run: zip --junk-paths ${{ matrix.platform }} ./release/${{ matrix.platform }}/*
-
-      ## Upload the binary artifact to the github action (used in `github-release.yml` to create a release)
-      - name: Upload artifact
-        id: upload_artifact
-        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+      - name: Build Binary (${{ matrix.arch }}_${{ matrix.cpu }})
+        id: build_binary
+        timeout-minutes: ${{ fromJSON(env.TEST_TIMEOUT) }}
```

Comment:
Why not just use `${{ env.TEST_TIMEOUT }}` here?

Reply:
@BowTiedDevOps  I would argue this env var shouldn't even be here. 
if a timeout is desired, it should be present in the build step of the composite action. 

i've never seen a timeout from a build though (only on tests where it's stuck in a loop)  so i'm inclined to say this should be removed.